packages/babylon/src/parser/statement.js

```diff
@@ -1422,6 +1425,12 @@ export default class StatementParser extends ExpressionParser {
     } else if (this.match(tt._class)) {
       return this.parseClass(expr, true, true);
     } else if (this.match(tt.at)) {
+      if (
+        this.hasPlugin("decorators") &&
+        this.getPluginOption("decorators", "decoratorsBeforeExport")
+      ) {
+        this.unexpected();
```

Comment:
It would be good to have these throw more helpful errors about what the user is supposed to do to get their code to parse.